pkg/command/map_string_test.go

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 Authors of Cilium
+
+//go:build !privileged_tests
+// +build !privileged_tests
+
+package command
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetStringMapString(t *testing.T) {
+	expectedResult := map[string]string{
+		"k1": "v1",
+		"k2": "v2",
+	}
+	type args struct {
+		key   string
+		value string
+	}
+	tests := []struct {
+		name string
+		args args
+		want map[string]string
+	}{
+		{
+			name: "value with json format",
+			args: args{
+				key:   "FOO_BAR",
+				value: `{"k1":"v1","k2":"v2"}`,
+			},
+			want: expectedResult,
+		},
+		{
+			name: "value with invalid json format",
+			args: args{
+				key:   "FOO_BAR",
+				value: `{"k1":"v1","k2":"v2",}`,
+			},
+			want: map[string]string{},
+		},
```

Comment:
in the error case like this i would usually check the error message for `strings.Contains` something specific like `bad json` or `extra comma`, I'm not sure what sort of a message is printed here.

Reply:
good point, I see the actual error in debugger only, making sense to make it explicit so that others don't need to re-check it. Let me refactor a little bit, and address related comments below as well.